pkg/resourcemanager/controller/tokenrequestor/reconciler_test.go

```diff
@@ -0,0 +1,380 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/client-go/kubernetes/scheme"
+	corev1fake "k8s.io/client-go/kubernetes/typed/core/v1/fake"
+	"k8s.io/client-go/testing"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("Reconciler", func() {
+	Describe("#Reconcile", func() {
+		var (
+			ctx = context.TODO()
+
+			logger     logr.Logger
+			fakeClock  clock.Clock
+			fakeJitter func(time.Duration, float64) time.Duration
+
+			sourceClient, targetClient client.Client
+			coreV1Client               *corev1fake.FakeCoreV1
+
+			ctrl *reconciler
+
+			secret         *corev1.Secret
+			serviceAccount *corev1.ServiceAccount
+			request        reconcile.Request
+
+			secretName              string
+			serviceAccountName      string
+			serviceAccountNamespace string
+			expectedRenewDuration   time.Duration
+			token                   string
+			fakeNow                 time.Time
+
+			fakeCreateServiceAccountToken = func() {
+				coreV1Client.AddReactor("create", "serviceaccounts", func(action testing.Action) (bool, runtime.Object, error) {
+					if action.GetSubresource() != "token" {
+						return false, nil, fmt.Errorf("subresource should be 'token'")
+					}
+
+					cAction, ok := action.(testing.CreateAction)
+					if !ok {
+						return false, nil, fmt.Errorf("could not convert action (type %T) to type testing.CreateAction", cAction)
+					}
+
+					tokenRequest, ok := cAction.GetObject().(*authenticationv1.TokenRequest)
+					if !ok {
+						return false, nil, fmt.Errorf("could not convert object (type %T) to type *authenticationv1.TokenRequest", cAction.GetObject())
+					}
+
+					return true, &authenticationv1.TokenRequest{
+						Status: authenticationv1.TokenRequestStatus{
+							Token:               token,
+							ExpirationTimestamp: metav1.Time{Time: fakeNow.Add(time.Duration(*tokenRequest.Spec.ExpirationSeconds) * time.Second)},
+						},
+					}, nil
+				})
+			}
+		)
+
+		BeforeEach(func() {
+			logger = log.Log.WithName("test")
+			fakeNow = time.Date(2021, 10, 4, 10, 0, 0, 0, time.UTC)
+			fakeClock = clock.NewFakeClock(fakeNow)
+			fakeJitter = func(d time.Duration, _ float64) time.Duration { return d }
+
+			sourceClient = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+			targetClient = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+			coreV1Client = &corev1fake.FakeCoreV1{Fake: &testing.Fake{}}
+
+			ctrl = &reconciler{
+				clock:              fakeClock,
+				jitter:             fakeJitter,
+				targetClient:       targetClient,
+				targetCoreV1Client: coreV1Client,
+			}
+
+			Expect(ctrl.InjectLogger(logger)).To(Succeed())
+			Expect(ctrl.InjectClient(sourceClient)).To(Succeed())
```

Comment:
```suggestion
			Expect(inject.LoggerInto(logger, ctrl)).To(Succeed())
			Expect(inject.ClientInto(sourceClient, ctrl)).To(Succeed())
```

I think, it's better to call the `inject.*Into` funcs instead (like the library does as well).
Maybe you can cleanup [this test](https://github.com/gardener/gardener/blob/65288a503da25ef2d89a0d019b4f19f78606986f/extensions/pkg/predicate/predicate_test.go#L203-L205) as well?